edgelet/iotedge-proxy/src/proxy/client.rs

```diff
@@ -94,6 +87,31 @@ where
     }
 }
 
+fn build_uri(destination_url: Url, requested_uri: &Uri) -> Result<Uri, Error> {
+    let path = percent_decode(requested_uri.path().as_bytes())
+        .decode_utf8()
+        .context(ErrorKind::Uri(requested_uri.to_string()))?;
```

Comment:
`.with_context`, because you don't want to run `.to_string()` eagerly.

Same for the next two calls.

Reply:
Good idea!